tst/utils/Thread.cpp

```diff
@@ -26,9 +26,10 @@ PVOID testThreadRoutine(PVOID arg)
 
     // Mark as visited
     st->threadVisited = TRUE;
+    UINT64 sleepTime = st->threadSleepTime;
```

Comment:
Moved this into the guarded section because:

```
2024-11-09T05:39:36.6508953Z ==================
2024-11-09T05:39:36.6509339Z WARNING: ThreadSanitizer: data race (pid=3464)
2024-11-09T05:39:36.6510114Z   Write of size 8 at 0x7ffc74c07ea0 by main thread:
2024-11-09T05:39:36.6512135Z     #0 defaultCreateThreadWithParams /home/runner/work/amazon-kinesis-video-streams-pic/amazon-kinesis-video-streams-pic/src/utils/src/Thread.c:210:16 (kvspic_test+0x3ea736) (BuildId: baf5dc344c10bb4b6e35338f8c13871da51827f4)
2024-11-09T05:39:36.6528552Z     #1 ThreadFunctionalityTest_ThreadCreateAndReleaseSimpleCheckWithStack_Test::TestBody() /home/runner/work/amazon-kinesis-video-streams-pic/amazon-kinesis-video-streams-pic/tst/utils/Thread.cpp:146:9 (kvspic_test+0x32e13e) (BuildId: baf5dc344c10bb4b6e35338f8c13871da51827f4)
2024-11-09T05:39:36.6530810Z     #2 void testing::internal::HandleSehExceptionsInMethodIfSupported<testing::Test, void>(testing::Test*, void (testing::Test::*)(), char const*) <null> (kvspic_test+0x44558a) (BuildId: baf5dc344c10bb4b6e35338f8c13871da51827f4)
2024-11-09T05:39:36.6532067Z     #3 <null> <null> (libc.so.6+0x29d8f) (BuildId: 490fef8403240c91833978d494d39e537409b92e)
2024-11-09T05:39:36.6532483Z 
2024-11-09T05:39:36.6532689Z   Previous read of size 4 at 0x7ffc74c07ea0 by thread T2752:
2024-11-09T05:39:36.6534085Z     #0 testThreadRoutine(void*) /home/runner/work/amazon-kinesis-video-streams-pic/amazon-kinesis-video-streams-pic/tst/utils/Thread.cpp:31:28 (kvspic_test+0x32cebc) (BuildId: baf5dc344c10bb4b6e35338f8c13871da51827f4)
2024-11-09T05:39:36.6535127Z 
2024-11-09T05:39:36.6535238Z   As if synchronized via sleep:
2024-11-09T05:39:36.6535757Z     #0 usleep <null> (kvspic_test+0xd1d2a) (BuildId: baf5dc344c10bb4b6e35338f8c13871da51827f4)
2024-11-09T05:39:36.6538617Z     #1 defaultThreadSleep /home/runner/work/amazon-kinesis-video-streams-pic/amazon-kinesis-video-streams-pic/src/utils/src/Thread.c:283:13 (kvspic_test+0x3ea93c) (BuildId: baf5dc344c10bb4b6e35338f8c13871da51827f4)
2024-11-09T05:39:36.6542904Z     #2 ThreadFunctionalityTest_ThreadCreateAndCancel_Test::TestBody() /home/runner/work/amazon-kinesis-video-streams-pic/amazon-kinesis-video-streams-pic/tst/utils/Thread.cpp:106:5 (kvspic_test+0x32da4c) (BuildId: baf5dc344c10bb4b6e35338f8c13871da51827f4)
2024-11-09T05:39:36.6545845Z     #3 void testing::internal::HandleSehExceptionsInMethodIfSupported<testing::Test, void>(testing::Test*, void (testing::Test::*)(), char const*) <null> (kvspic_test+0x44558a) (BuildId: baf5dc344c10bb4b6e35338f8c13871da51827f4)
2024-11-09T05:39:36.6547095Z     #4 <null> <null> (libc.so.6+0x29d8f) (BuildId: 490fef8403240c91833978d494d39e537409b92e)
2024-11-09T05:39:36.6547524Z 
2024-11-09T05:39:36.6547641Z   Location is stack of main thread.
2024-11-09T05:39:36.6547876Z 
2024-11-09T05:39:36.6548143Z   Location is global '??' at 0x7ffc74beb000 ([stack]+0x1cea0)
2024-11-09T05:39:36.6548460Z 
2024-11-09T05:39:36.6548662Z   Thread T2752 (tid=6243, finished) created by main thread at:
2024-11-09T05:39:36.6549329Z     #0 pthread_create <null> (kvspic_test+0xd478d) (BuildId: baf5dc344c10bb4b6e35338f8c13871da51827f4)
2024-11-09T05:39:36.6550946Z     #1 defaultCreateThreadWithParams /home/runner/work/amazon-kinesis-video-streams-pic/amazon-kinesis-video-streams-pic/src/utils/src/Thread.c:194:14 (kvspic_test+0x3ea6fd) (BuildId: baf5dc344c10bb4b6e35338f8c13871da51827f4)
2024-11-09T05:39:36.6553202Z     #2 defaultCreateThread /home/runner/work/amazon-kinesis-video-streams-pic/amazon-kinesis-video-streams-pic/src/utils/src/Thread.c:242:5 (kvspic_test+0x3ea84f) (BuildId: baf5dc344c10bb4b6e35338f8c13871da51827f4)
2024-11-09T05:39:36.6555542Z     #3 ThreadFunctionalityTest_ThreadCreateAndCancel_Test::TestBody() /home/runner/work/amazon-kinesis-video-streams-pic/amazon-kinesis-video-streams-pic/tst/utils/Thread.cpp:96:9 (kvspic_test+0x32d7fc) (BuildId: baf5dc344c10bb4b6e35338f8c13871da51827f4)
2024-11-09T05:39:36.6557567Z     #4 void testing::internal::HandleSehExceptionsInMethodIfSupported<testing::Test, void>(testing::Test*, void (testing::Test::*)(), char const*) <null> (kvspic_test+0x44558a) (BuildId: baf5dc344c10bb4b6e35338f8c13871da51827f4)
2024-11-09T05:39:36.6558795Z     #5 <null> <null> (libc.so.6+0x29d8f) (BuildId: 490fef8403240c91833978d494d39e537409b92e)
2024-11-09T05:39:36.6559226Z 
2024-11-09T05:39:36.6560158Z SUMMARY: ThreadSanitizer: data race /home/runner/work/amazon-kinesis-video-streams-pic/amazon-kinesis-video-streams-pic/src/utils/src/Thread.c:210:16 in defaultCreateThreadWithParams
2024-11-09T05:39:36.6561185Z ==================
2024-11-09T05:39:36.6864834Z ##[error]Process completed with exit code 66.
```